File: StudentManagement/DTOs/User/UserCreateDto.cs
using System.ComponentModel.DataAnnotations;

namespace StudentManagementSystem.Dtos.User
{
    public class UserCreateDto
    {
        [Required]
        [StringLength(10)]
        public string UserId { get; set; } = string.Empty;

        [Required]
        [StringLength(50)]
        public string Username { get; set; } = string.Empty;

        [Required]
        [StringLength(100)]
        [EmailAddress]
        public string Email { get; set; } = string.Empty;

        [Required]
        [StringLength(255)]
        public string Password { get; set; } = string.Empty;

        [Required]
        [StringLength(10)]
        public string RoleId { get; set; } = string.Empty;

        [StringLength(10)]
        public string? StudentId { get; set; }

        [StringLength(10)]
        public string? EmployeeId { get; set; }

        [StringLength(10)]
        public string? TeacherId { get; set; }

        public bool IsActive { get; set; } = true;
    }
}

File: StudentManagement.Web/src/app/models/dtos/user-create.dto.ts
export interface UserCreateDto {
  userId: string;
  username: string;
  email: string;
  password?: string; // Mật khẩu là bắt buộc khi tạo mới
  roleId: string;
  studentId?: string | null;
  employeeId?: string | null;
  teacherId?: string | null;
  isActive: boolean;
}
